scripts/build

```diff
@@ -5,13 +5,26 @@ TOP_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." &> /dev/null && pwd )"
 SCRIPTS_DIR="${TOP_DIR}/scripts"
 PACKAGE_HARVESTER_OS_DIR="${TOP_DIR}/package/harvester-os"
 
-source ${SCRIPTS_DIR}/version-rke2
-
 cd ${TOP_DIR}
 
+# Pull harvester source and determine harvester version
+harvester_path=../harvester
+if [ ! -d ${harvester_path} ];then
+    echo "No existed harvester source. Pulling..."
+    git clone --branch master --single-branch --depth 1 https://github.com/harvester/harvester.git ../harvester
+fi
+HARVESTER_VERSION=$(${SCRIPTS_DIR}/version-harvester $harvester_path)
+echo "Harvester version: $HARVESTER_VERSION"
```

Comment:
Can we move the command substitution to version-harvester so that we can do `source ${SCRIPTS_DIR}/version-harvester`

Reply:
`version-harvester` needs to source harvester's `version` script to get version information, which would also introduce other variables like `GIT_TAG` and `COMMIT` into the current scope. I'm a bit worried about this because the **harvester-installer**'s `version` script also defines variables with identical names, and this may cause confusion when we `source` both scripts into the current context. By executing `version-harvester` instead of `source`, only the variable we want i.e. the `VERSION` variable would be "returned" and we could assign it to a different variable.

Nevertheless, I did it this way to avoid changing too many things altogether in harvester-installer and harvester repositories. If we prefer the `source` way, I could totally make a change ;)

Reply:
Example:
```
# version-harvester
HARVESTER_VERSION=$(source $harvester_path/scripts/version;echo $VERSION)

# scripts/build
source ${SCRIPTS_DIR}/version-harvester
```
The idea is to make the `version-harvester` script self-contained so that the caller can simply source it. There's no much difference from current implementation.